src/bsoncxx/string/view_or_value.cpp

```diff
@@ -15,20 +15,26 @@
 #include <bsoncxx/config/prelude.hpp>
 
 #include <bsoncxx/string/view_or_value.hpp>
+#include <bsoncxx/exception/error_code.hpp>
+#include <bsoncxx/exception/exception.hpp>
 
 namespace bsoncxx {
 BSONCXX_INLINE_NAMESPACE_BEGIN
 namespace string {
 
-const char* view_or_value::c_str() {
+view_or_value view_or_value::terminated() const {
     // If we do not own our string, we cannot guarantee that it is null-terminated,
     // so make an owned copy.
-    if (!_value) {
-        _value = std::move(_view.to_string());
-        _view = *_value;
+    if (!is_owning()) {
+        return view_or_value(std::move(view().to_string()));
```

Comment:
I think the std::move here is unneeded as the result of `to_string` is a temporary already, but having thought about it, I think I like it for clarity.